File: benchmarks/language_compare/arithmetic.c
/* BENCH_ITERATIONS=5000000 BENCH_EXPECTED=497907426 */
#include <stdint.h>
#include <stdio.h>

int main(void) {
    int64_t state = 1;
    int64_t checksum = 0;
    for (int64_t i = 0; i < 5000000; i++) {
        state = (state * 48271) % 2147483647;
        checksum = (checksum + state % 1000) % 1000000007;
    }
    printf("%lld\n", (long long)checksum);
    return 0;
}

File: benchmarks/language_compare/arithmetic.rs
// BENCH_ITERATIONS=5000000 BENCH_EXPECTED=497907426
fn main() {
    let mut state: i64 = 1;
    let mut checksum: i64 = 0;
    for _ in 0..5_000_000 {
        state = (state * 48_271) % 2_147_483_647;
        checksum = (checksum + state % 1_000) % 1_000_000_007;
    }
    println!("{checksum}");
}
